### ISSUE TITLE
<client> with empty inet object throws bad error

### ISSUE THREAD

**Issue by erg** [factor/factor#2109]
```factor
T{ inet f f f } ascii <client>

Bad store to specialized slot
value
f
class
integer
```

**Comment by mrjbq7** [comment]
Weird that the documentation says ``f`` as a port defers choice until later, but the definition of ``inet4`` requires an integer, defaulting to ``0``.

```factor
IN: scratchpad "127.0.0.1" f <inet4>

Bad store to specialized slot
value
f
class
integer
```

**Comment by mrjbq7** [comment]
@erg do you think we should change the slot to ``maybe{ integer }`` or should we change the docs to remove the comment about "*port of ``f`` defers the port choice until later*".

**Comment by erg** [comment]
For servers, 0/f makes sense, and now both work (again?)
```factor
"127.0.0.1" 0 <inet4> utf8 <server>
"127.0.0.1" f <inet4> utf8 <server>
```

**Comment by erg** [comment]
Fixed by f4d17d3ba394c4bac21ad800c2d5d05fb443d5ad and d59292faf68094dbc219aeb7ff63947423c21fe6
